### PR TITLE
Switch 64-bit hashes to xxhash3

### PR DESCRIPTION
**Pull request by grendello**
Performance improvements compared to `main`, tests with Android 14 on Pixel 6 Pro:

   * **Displayed time**: 2.17% faster
   * **Native-to-managed init**: 1.4% faster
   * **Total native init**: 1.15% faster

### CODE REVIEW

**Comment by jpobst** [comment]
I don't know if we only use this from managed code, but I recently saw that the .NET Team maintains a NuGet with these algorithms:
https://www.nuget.org/packages/System.IO.Hashing#readme-body-tab

**Comment by grendello** [comment]
@jpobst we use the `System.IO.Hashing` nuget, that's where the xxHash3 APIs are from.  We also use the algorithm at runtime in native code, but we have native implementation(s) there instead.

**Review by jonathanpeppers — APPROVED** [review]
The managed side looks ok to me, we already ship `System.IO.Hashing.dll` so should be fine to use:

* https://learn.microsoft.com/dotnet/api/system.io.hashing.xxhash3

(Probably other review needed for the native code)

File: Documentation/guides/AndroidAssetPacks.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+# Android Asset Packs
```
Comment (dellis1972) [inline Documentation/guides/AndroidAssetPacks.md:1]: Err this was merged into main? I wonder why its appearing in this PR?

**Comment by jonpryor** [comment]
@grendello: please also update the `THIRD-PARTY-NOTICES.TXT` generation code to include these repositories and licenses.

**Pull request merged**